dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +111,234 @@ public Bundle getBundleById(String id) throws DotDataException {
+				this.bundleFactory.findSentBundles(olderThan, userId, limit, offset);
+		BufferedWriter fileWriter = null;
+
+		// if the initial is less than limit, return it
+		if (UtilMethods.isSet(sentBundles) && sentBundles.size() < limit) {
+
+			return sentBundles.stream().map(Bundle::getId);
+		}
+
+		try {
+
+			fileWriter = new BufferedWriter(new FileWriter(tempFile));
+			while (UtilMethods.isSet(sentBundles)) {
+
+				for (final Bundle bundle : sentBundles) {
+
+					fileWriter.write(bundle.getId());
+					fileWriter.newLine();
+				}
+
+				fileWriter.flush();
+
+				offset       += limit + 1;
+				sentBundles   = isAdmin?
+						this.bundleFactory.findSentBundles(olderThan, limit, offset):
+						this.bundleFactory.findSentBundles(olderThan, userId, limit, offset);
+			}
+		} finally {
+
+			CloseUtils.closeQuietly(fileWriter);
+		}
+
+		return Files.lines(tempFile.toPath());
+	}
+
+	private void internalDeleteBundleAndDependencies(final String bundleId, final User user) {
+
+		try {
+
+			this.deleteBundleAndDependencies(bundleId, user);
+		} catch (DotDataException e) {
+
+			throw new DotRuntimeException(e);
+		}
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String> deleteAllBundles(final User user,
+										 final PublishAuditStatus.Status ...statuses) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final int limit          = 100;
+		final int offset         = 0;
+		final boolean isAdmin    = this.userAPI.isCMSAdmin(user);
+		final List<Status> statusList = Arrays.asList(statuses);
+		List<String> bundleIDs = isAdmin ? publishAuditAPI.getBundleIdByStatus(statusList,limit,offset) :
+				publishAuditAPI.getBundleIdByStatusFilterByOwner(statusList,limit,offset,user.getUserId());
+//		List<Bundle> sentBundles = isAdmin?
+//				this.bundleFactory.findSentBundles(limit, offset): // todo: modify this method in order to use publishing_queue_audit,
+//				this.bundleFactory.findSentBundles(user.getUserId(), limit, offset);
+
+		Logger.info(this, "Deleting all bundles with statuses: "
+				+ Arrays.asList(statuses));
+		while (UtilMethods.isSet(sentBundles)) {
+
+			for (final Bundle bundle : sentBundles) {
+
+				try {
+
+					final PublishAuditStatus status =
+							this.publishAuditAPI.getPublishAuditStatus(bundle.getId());
+
+					if (this.containsStatus (status, statuses)) {
+
+						this.deleteBundleAndDependencies(bundle, user);
+						bundlesDeleted.add(bundle.getId());
+					}
+				} catch (final DotPublisherException e) {
+
+					throw new DotDataException(e);
+				}
+			}
+
+			sentBundles = isAdmin?
+					this.bundleFactory.findSentBundles(limit, offset):
+					this.bundleFactory.findSentBundles(user.getUserId(), limit, offset);
+		}
+
+		return bundlesDeleted.build();
+	} // deleteAllBundles.
+
+	private boolean containsStatus(final PublishAuditStatus          bundleStatus,
+								   final PublishAuditStatus.Status[] statuses) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)